src/api/objects/LegacyObjectAPIInterceptor.js

```diff
@@ -1,15 +1,56 @@
 define([
     './object-utils',
-    './ObjectAPI'
+    './ObjectAPI',
+    './objectEventEmitter'
 ], function (
     utils,
-    ObjectAPI
+    ObjectAPI,
+    objectEventEmitter
 ) {
-    function ObjectServiceProvider(objectService, instantiate) {
+    function ObjectServiceProvider(objectService, instantiate, topic) {
         this.objectService = objectService;
         this.instantiate = instantiate;
+
+        this.topicService = topic;
+        this.generalTopic = topic('mutation');
+        this.bridgeEventBuses();
     }
 
+    /**
+     * Bridges old and new style mutation events to provide compatibility between the two APIs
+     * @private
+     */
+    ObjectServiceProvider.prototype.bridgeEventBuses = function () {
+        var generalTopicListener;
+
+        var handleMutation = function (newStyleObject) {
+            var keyString = utils.makeKeyString(newStyleObject.key);
+            var oldStyleObject = this.instantiate(utils.toOldFormat(newStyleObject), keyString);
+            var specificTopic = this.topicService("mutation:" + keyString);
+
+            // Don't trigger self
+            if (generalTopicListener){
```

Comment:
It's impossible for `generalTopicListener` to be falsy, so best to remove this check to ensure no unexpected code paths.

This is a good place for a comment, but it could be clearer, e.g. "temporarily remove listener to prevent infinite mutation loop".  Might also rename the variable to something like `removeGeneralTopicListener`.